### PR TITLE
Allow setting options on the DB

### PR DESCRIPTION
**Pull request by romanz**
This should allow changing options (e.g. `disable_auto_compactions`) without closing and re-opening the database.

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -1123,6 +1123,17 @@ impl DB {
             );
         }
     }
+
+    pub fn set_option(&self, name: &str, value: &str) -> Result<(), Error> {
+        let cname = CString::new(name.as_bytes()).unwrap();
```
Comment (aleksuss) [inline src/db.rs]: To use `unwrap()` here it's a bad idea. 
```suggestion
         let cname = CString::new(name.as_bytes())?;
```

File: src/db.rs [hunk 2]
```diff
@@ -1497,3 +1508,15 @@ fn snapshot_test() {
...
 }
+
+#[test]
+fn set_option_test() {
+    let path = "_rust_rocksdb_set_optionstest";
+    {
+        let db = DB::open_default(path).unwrap();
+        assert!(db.set_option("disable_auto_compactions", "true").is_ok());
+        assert!(db.set_option("disable_auto_compactions", "false").is_ok());
+        assert!(db.set_option("disable_auto_compactions", "INVALID_VALUE").is_err());
```
Comment (aleksuss) [inline src/db.rs]: What about other options or this implementation related only for `disable_auto_compactions` option ?

File: src/db.rs [hunk 3]
```diff
@@ -1123,6 +1123,17 @@ impl DB {
             );
         }
     }
+
+    pub fn set_option(&self, name: &str, value: &str) -> Result<(), Error> {
```
Comment (aleksuss) [inline src/db.rs]: Have you thought about to make this method similar to original function? What if I want to change more than one parameter ? The implementation as I can see use synchronisation to apply this changes in runtime. So this is not a cheap operation.

File: src/db.rs (hunk 2 above)
Comment (romanz) [reply src/db.rs]: This should also work for other options as well (IIUC, the list is [here](https://github.com/facebook/rocksdb/blob/46e3209e0da1d7563e345d0e28a82ec05f1ddf32/options/options_helper.cc#L1663)). Currently in [electrs](https://github.com/romanz/electrs), I am planning using it for setting `disable_auto_compactions` to `false` after the initial bulk load is over.

**Comment by romanz** [comment]
Thanks for the review :)
Answered the comments inline, and force-pushed 73aa6bf.

**Comment by iSynaptic** [comment]
@aleksuss Was all of your feedback addressed?

**Comment by romanz** [comment]
Thanks!

**Pull request merged**